### ISSUE TITLE
when use bayes tuning xgboost, is there need to set scale_pos_weight ?

### ISSUE THREAD

**Issue by StutuPig** [Microsoft/LightGBM#1058]
if neg/pos is about 45-65, in this case, should we set scale_pos_weight to 1 or neg/pos ?
when i use BayesianOptimization(bayes tuning) to tune xgboost, if i set scale_pos_weight to neg/pos the ks on valid dataset is about 18, but if i set scale_pos_weight to 1, the ks on valid dataset up to 38 ... i don't know why ? can somebody help to explain ..
(Note: if scale_pos_weight ==1, set feval='ks', or else, set feval='auc')

Please search your question on previous issues, [stackoverflow](https://stackoverflow.com/questions/tagged/lightgbm) or other search engines before you open a new one.

For bugs and unexpected issues, please provide following information, so that we could reproduce on our system.

## Environment info
Operating System:
CPU:
C++/Python/R version:

## Steps to reproduce

1.
2.
3.

**Comment by StrikerRUS** [comment]
XGBoost? This is LightGBM's repo, ask them in their repo: https://github.com/dmlc/xgboost

**Comment by Laurae2** [comment]
You need to take into account in xgboost the scale_pos_weight scales not only the gradients/hessians, but also the evaluation metric.

**Comment by Laurae2** [comment]
@StrikerRUS You can close this issue (I can't from my phone)
